File: Programs/Go/division.go
package main

import "fmt"

func division (a int, b int) int {
    if a < b{
        return 0
    }else{
        return 1 + division(a-b, b)
    }
}

func main(){
    a, b := 8,2
    fmt.Printf("El resultado de la division entre %g y % g es: %g",a,b, division(a, b))
}

File: Programs/C/division.c
#include <stdio.h>

int division(int x, int y){
    if (x<y)
        return 0;
    else
        return 1 + division(x-y,y);
}

int main(){
    int x, y;
    printf("ingrese el dividendo: \n");
    scanf("%d",&x);
    printf("Ingrese el divisor: \n");
    scanf("%d",&y);
    printf("El resultado de la división entera entre %d y %d es de: %d", x, y, division(x,y));
}

Hola que 
